pkg/tracing/interceptors/reporter.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 Michal Witkowski. All Rights Reserved.
```

Comment:
Just for clarification: 

We are doing this temporarily until we migrate Thanos Tracing to OTEL provider, instead of OpenTracing. I do believe most of them are OTEL compatible now, but it was a good idea to split it up.